src/memray/_memray/tracking_api.cpp

```diff
@@ -388,6 +388,25 @@ PythonStackTracker::handleGreenletSwitch(PyObject*, PyObject*)
         emitPendingPushesAndPops();
     }
 
+    // Save current TID on old greenlet. Print errors but otherwise ignore them.
+    PyObject* tid = PyLong_FromUnsignedLong(t_tid);
+    if (!tid || 0 != PyObject_SetAttrString(from, "_memray_tid", tid)) {
+        PyErr_Print();
```

Comment:
Don't we want to return/abort in this case? Is technically not a catastrophic failure, but technically will produce incorrect results when switching back to, no?